src/ndarray/ndarray_function.cc

```diff
@@ -42,15 +43,18 @@ void ElementwiseSumRspImpl(const std::vector<NDArray>& nds,
     if (row_block_start < nnr) {
       const size_t row_block_end = std::min(row_block_start+row_block_len, nnr);
 
-      auto out_values = out->data().FlatTo2D<cpu, DType>();
+      const size_t num_cols = out->data().shape_.ProdShape(1, out->data().shape_.ndim());
+      auto out_values = out->data().get_with_shape<cpu, 2, DType>(
+          mshadow::Shape2(out->storage_shape()[0], num_cols), s);
       auto out_indices = out->aux_data(rowsparse::kIdx).FlatTo1D<cpu, IType>();
       for (size_t i = row_block_start; i < row_block_end; ++i) {
         out_indices[i] = uniq_row_idx[i];
       }
       for (const auto& nd : nds) {
         if (nd.storage_initialized()) {
           const auto nd_indices = nd.aux_data(rowsparse::kIdx).FlatTo1D<cpu, IType>();
-          const auto nd_values = nd.data().FlatTo2D<cpu, DType>();
+          const auto nd_values = nd.data().get_with_shape<cpu, 2, DType>(
```

Comment:
He suggested not adding a bool argument but calculating the desired shape explicitly.